static/app/stores/Commiters/CommittersContext.tsx

```diff
@@ -0,0 +1,274 @@
+        organizationSlug: action.payload.organizationSlug,
+        projectSlug: action.payload.projectSlug,
+        eventId: action.payload.eventId,
+      });
+
+      return {
+        ...state,
+        [key]: {
+          committers: action.payload.committers,
+          committersLoading: false,
+          committersError: undefined,
+        },
+      };
+    }
+
+    case 'start loading': {
+      const key = makeCommitterStoreKey({
+        organizationSlug: action.payload.organizationSlug,
+        projectSlug: action.payload.projectSlug,
+        eventId: action.payload.eventId,
+      });
+      return {
+        ...state,
+        [key]: {
+          committers: [],
+          committersLoading: true,
+          committersError: undefined,
+        },
+      };
+    }
+    case 'set error': {
+      const key = makeCommitterStoreKey({
+        organizationSlug: action.payload.organizationSlug,
+        projectSlug: action.payload.projectSlug,
+        eventId: action.payload.eventId,
+      });
+
+      return {
+        ...state,
+        [key]: {
+          committers: [],
+          committersLoading: false,
+          committersError: true,
+        },
+      };
+    }
+    case 'reset': {
+      return {};
+    }
+    default: {
+      exhaustive(action);
+      return state;
+    }
+  }
+}
+
+export const CommittersContext = createContext<
+  [CommittersState, React.Dispatch<CommittersAction>] | null
+>(null);
+
+interface CommittersContextProviderProps {
+  children: React.ReactNode;
+  initialState?: CommittersState;
+}
+
+export function CommittersProvider(props: CommittersContextProviderProps) {
+  const contextValue = useReducer(committersReducer, props.initialState ?? {});
+
+  return (
+    <CommittersContext.Provider value={contextValue}>
+      {props.children}
+    </CommittersContext.Provider>
+  );
+}
+
+export function useCommitters(): [CommittersState, React.Dispatch<CommittersAction>] {
+  const context = useContext(CommittersContext);
+
+  if (!context) {
+    throw new Error('useCommitters called outside of CommittersContext.Provider');
+  }
+
+  return context;
+}
+
+interface RequiredWithCommittersProps {
+  event: Event;
+  organization: Organization;
+  project: Project | AvatarProject;
+  group?: Group;
+}
+
+export interface WithCommittersProps {
+  committers: Committer[] | undefined;
+}
+
+export function withCommitters<P extends WithCommittersProps>(
+  Component: React.ComponentType<P>
+) {
+  const wrappedComponent: React.ComponentType<
+    Omit<P, keyof WithCommittersProps> & RequiredWithCommittersProps
+  > = (props): React.ReactElement => {
+    const api = useApi();
+    const [state, dispatch] = useCommitters();
+
+    const key = useMemo(() => {
+      return makeCommitterStoreKey({
+        organizationSlug: props.organization.slug,
+        projectSlug: props.project.slug,
+        eventId: props.event.id,
+      });
+    }, [props.organization.slug, props.project.slug, props.event.id]);
```

Comment:
Won't the memoization add more overhead than calling the key generation function on each render?

Reply:
Probably, I havent really thought about it in those terms tbh, I just wanted to show that it's somewhat "stable"